spec/03-plugins/25-oauth2/03-access_spec.lua

```diff
@@ -2905,6 +2935,76 @@ describe("Plugin: oauth2 [#" .. strategy .. "]", function()
         })
         assert.res_status(401, res)
       end)
+      it("fails to refresh token when scope is mismatching", function ()
+        -- provision code
```

Comment:
Style: I think instead of using code flow, using others to get an access_token/refresh_token can make the test simpler.

Reply:
Yeah, however I have noticed ahead of time that there are a few funcs which need some modifications to be usable to do the same matters. I suppose it might not worth of making the test simpler with the cost of more changes.